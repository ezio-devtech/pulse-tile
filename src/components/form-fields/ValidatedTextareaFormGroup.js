import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default class ValidatedTextareaFormGroup extends PureComponent {
  static propTypes = {
    label: PropTypes.string.isRequired,
    input: PropTypes.object.isRequired,
    meta: PropTypes.shape({
      active: PropTypes.bool,
      error: PropTypes.string,
    }).isRequired,
  };

  state={
    isChanged: false,
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.meta.dirty) {
      this.setState({ isChanged: true })
    }
  }

  render() {
    const { label, input, meta: { error, touched }, id, isSubmit } = this.props;
    const { isChanged } = this.state;
    const showError = ((touched || isChanged || isSubmit) && error);

    return (
      <div className={classNames('form-group', { 'has-error': showError }, { 'has-success': isChanged && !error })}>
        <label htmlFor={input.name} className="control-label">{label}</label>
        <div className="input-holder">
          <textarea
            className="form-control textarea-big input-sm ng-pristine ng-empty ng-invalid ng-invalid-required ng-touched"
            id={id}
            {...input}
          />
        </div>
        {showError && <span className="required-label">{error}</span>}
      </div>
    )
  }
}
