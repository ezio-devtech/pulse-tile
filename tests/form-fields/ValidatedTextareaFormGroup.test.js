import React from 'react';
import {configure, shallow, mount} from 'enzyme'
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });

import ValidatedTextareaFormGroup from '../../src/components/form-fields/ValidatedTextareaFormGroup';

const testProps = {
  label: 'Test label',
  input: {name: 'test-name', value: ''},
  inputAndImgSRC: {name: 'test-name', value: 'image.jpg'},
  meta: {
    active: false,
    error: false,
  },
  id: 'test-id',
};

describe('Component <ValidatedTextareaFormGroup />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
        <ValidatedTextareaFormGroup
          id={testProps.id}
          label={testProps.label}
          input={testProps.input}
          meta={testProps.meta}
          isSubmit={true}
          isAdvancedSearce={false}
        />);

    expect(component.find('input')).toHaveLength(0);
    expect(component.find('textarea')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(1);

    expect(component.find('label.control-label').text()).toEqual(testProps.label);
    expect(component.find('label.control-label').prop('htmlFor')).toEqual(testProps.id);

    expect(component.find('textarea').prop('id')).toEqual(testProps.id);

    expect(component.instance().props['id']).toEqual(testProps.id);
    expect(component.instance().props['label']).toEqual(testProps.label);
    expect(component.instance().props['input']).toEqual(testProps.input);
    expect(component.instance().props['meta']).toEqual(testProps.meta);
    expect(component.instance().props['isSubmit']).toEqual(true);
    expect(component.instance().props['isAdvancedSearce']).toEqual(false);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  // it('should renders with image correctly', () => {
  //   const component = shallow(
  //     <ValidatedTextareaFormGroup
  //       id={testProps.id}
  //       label={testProps.label}
  //       input={testProps.inputAndImgSRC}
  //       meta={testProps.meta}
  //     />);
  //
  //   expect(component.find('img')).toHaveLength(1);
  //   expect(component.find('img').prop('src')).toEqual(testProps.inputAndImgSRC.value);
  //
  //   const tree = toJson(component);
  //   expect(tree).toMatchSnapshot();
  // });

  // it('should renders with different state of validation', () => {
  //   let tree;
  //   const component = shallow(
  //     <ValidatedTextareaFormGroup
  //       id={testProps.id}
  //       label={testProps.label}
  //       input={testProps.inputAndImgSRC}
  //       meta={testProps.meta}
  //     />);
  //
  //   expect(component.find('.form-group').at(0).hasClass('has-error')).toEqual(false);
  //   expect(component.find('.form-group').at(0).hasClass('has-success')).toEqual(false);
  //   tree = toJson(component);
  //   expect(tree).toMatchSnapshot();
  //
  //   component.setProps({ meta: {
  //     active: false,
  //     error: 'Error'
  //   }});
  //   expect(component.find('.form-group').at(0).hasClass('has-error')).toEqual(true);
  //   expect(component.find('.form-group').at(0).hasClass('has-success')).toEqual(false);
  //   tree = toJson(component);
  //   expect(tree).toMatchSnapshot();
  //
  //   component.setProps({meta: {
  //     active: true,
  //     error: false
  //   }});
  //   expect(component.find('.form-group').at(0).hasClass('has-error')).toEqual(false);
  //   expect(component.find('.form-group').at(0).hasClass('has-success')).toEqual(true);
  //   tree = toJson(component);
  //   expect(tree).toMatchSnapshot();
  // });
});
