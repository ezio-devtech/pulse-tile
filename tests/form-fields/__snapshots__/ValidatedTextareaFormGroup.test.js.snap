// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ValidatedTextareaFormGroup /> should renders with all props correctly 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="test-id"
  >
    Test label
  </label>
  <div
    className="input-holder"
  >
    <textarea
      className="form-control textarea-big input-sm"
      id="test-id"
      name="test-name"
      value=""
    />
  </div>
</div>
`;
